pkgs/games/mindustry/default.nix

```diff
@@ -81,66 +120,76 @@ let
     '';
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = "0vzck6hsrvs438s3ikk66qmpak88bmqcb8inqbbjwy7x87d2qsvj";
+    outputHash = "09rwyrg2yv8r499b0dk1bzvymsf98d4j5b95bwd9s4xvrz71is3l";
   };
 
-  # Separate commands for building and installing the server and the client
-  buildClient = ''
+in
+assert stdenv.lib.assertMsg (enableClient || enableServer)
+  "mindustry: at least one of 'enableClient' and 'enableServer' must be true";
+stdenv.mkDerivation rec {
+  inherit pname version unpackPhase;
+
+  prePatch = "cd ../Arc";
+
+  # patches for Arc
+  patches = [
+    ./0001-fix-include-path-for-SDL2-on-linux.patch
+    # upstream fix for https://github.com/Anuken/Arc/issues/40, remove on next release
+    (fetchpatch {
+      url = "https://github.com/Anuken/Arc/commit/b2f3d212c1a88a62f140f5cb04f4c86e61332d1c.patch";
+      sha256 = "1z0vxr906mmi8vk75s3xv6cbn2flv2pikkv5yifmiq1xywhc1gwz";
+    })
+  ];
+
+  postPatch = ''
+    rm natives/natives-desktop/libs/libarc*.so
+    rm backends/backend-sdl/libs/linux64/libsdl-arc*.so
+
+    cd ../Mindustry
+  '' + cleanupMindustrySrc;
+
+  buildInputs = [ SDL2 glew alsaLib ];
+  nativeBuildInputs = [ pkg-config gradle_6 makeWrapper jdk ant copyDesktopItems ];
```

Comment:
I like to have these on separate lines, so diffs are more granular

Reply:
Done.